sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateChangeLogDao.java

```diff
@@ -0,0 +1,160 @@
+package uk.ac.bbsrc.tgac.miso.persistence.impl;
+
+import java.util.Date;
+import java.util.List;
+
+import org.hibernate.Criteria;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.criterion.Restrictions;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+
+import uk.ac.bbsrc.tgac.miso.core.data.ChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.changelog.BoxChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.changelog.ExperimentChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.changelog.KitDescriptorChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.changelog.LibraryChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.changelog.PoolChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.changelog.RunChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.changelog.SampleChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.changelog.SequencerPartitionContainerChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.changelog.StudyChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.store.ChangeLogStore;
+
+@Transactional(rollbackFor = Exception.class)
+public class HibernateChangeLogDao implements ChangeLogStore {
+
+  @Autowired
+  private SessionFactory sessionFactory;
+
+  public void setSessionFactory(SessionFactory sessionFactory) {
+    this.sessionFactory = sessionFactory;
+  }
+
+  private Session currentSession() {
+    return sessionFactory.getCurrentSession();
+  }
+
+  @Override
+  public List<ChangeLog> listAll(String type) {
+    ChangeLogType changeLogType = ChangeLogType.get(type);
+    Criteria criteria = currentSession().createCriteria(changeLogType.getClazz());
+    @SuppressWarnings("unchecked")
+    List<ChangeLog> results = criteria.list();
+    return results;
+  }
+
+  @Override
+  public List<ChangeLog> listAllById(String type, long id) {
+    ChangeLogType changeLogType = ChangeLogType.get(type);
+    Criteria criteria = currentSession().createCriteria(changeLogType.getClazz());
+    criteria.createAlias("changeLog", "changeLog");
+    criteria.add(Restrictions.eq("changeLog.id", id));
+    @SuppressWarnings("unchecked")
+    List<ChangeLog> results = criteria.list();
+    return results;
+  }
+
+  @Override
+  public void deleteAllById(String type, long id) {
+    ChangeLogType changeLogType = ChangeLogType.get(type);
+    Query query = currentSession().createQuery(String.format("delete %s where id = :ID", changeLogType.getTableName()));
+    query.setParameter("ID", id);
+    query.executeUpdate();
+  }
+
+  @Override
+  public Long create(String type, long entityId, ChangeLog changeLog) {
```

Comment:
If all the changelogs are created via triggers, do we need a DAO method for creating them?